### PR TITLE
refactor: Migrate UndoRedoKeyListeners to typescript

### PR DESCRIPTION
**Pull request by EnxDev**
### SUMMARY

Migrate UndoRedoKeyListeners to typescript

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N.A

### TESTING INSTRUCTIONS

All tests should pass

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by geido — APPROVED** [review]

File: superset-frontend/src/dashboard/components/UndoRedoKeyListeners/index.tsx [hunk 1]
```diff
@@ -17,15 +17,15 @@
  * under the License.
  */
 import { PureComponent } from 'react';
-import PropTypes from 'prop-types';
+import { HeaderProps } from '../Header/types';
 
-const propTypes = {
-  onUndo: PropTypes.func.isRequired,
-  onRedo: PropTypes.func.isRequired,
+type UndoRedoKeyListenersProps = {
```
[the remark this answers is not in the document]
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/UndoRedoKeyListeners/index.tsx:22]: But your code is fine as well, so let me know if you'd rather keep it as it is and I'll merge the PR

**Review by kgabryje — APPROVED** [review]

File: superset-frontend/src/dashboard/components/UndoRedoKeyListeners/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (EnxDev) [reply superset-frontend/src/dashboard/components/UndoRedoKeyListeners/index.tsx:22]: > But your code is fine as well, so let me know if you'd rather keep it as it is and I'll merge the PR

If you’d like, I can modify it.

[the remark this answers is not in the document]
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/UndoRedoKeyListeners/index.tsx:22]: All good, merged 🙂

**Pull request merged**